app/models/shipit/undeployed_commit.rb

```diff
@@ -35,12 +35,30 @@ def deploy_disallowed?
     end
 
     def deploy_discouraged?
-      stack.maximum_commits_per_deploy && index >= stack.maximum_commits_per_deploy
+      maximum_commits_per_deploy_reached?
+    end
+
+    def deploy_scheduled?
+      stack.continuous_deployment && !maximum_commits_per_deploy_reached? && !active?
     end
 
     def blocked?
       return @blocked if defined?(@blocked)
       @blocked = super
     end
+
+    def active?
+      return false unless stack.active_task?
+
+      active_task = stack.active_task
+
+      id > active_task.since_commit.id && id <= active_task.until_commit.id
```

Comment:
\nit using the `between?` method could make sense here https://ruby-doc.org/core-2.2.0/Comparable.html#method-i-between-3F

Reply:
I thought about it but `between?` uses a `>=` comparison instead of `>`. To be honest I will review this code, I was thinking about it and I think it will fail when the project has just one commit, I will add some tests to make sure how it is going to behave.